### PR TITLE
ci: fetch collaborators from github

### LINKED ISSUE #402: Automate collaborators.yaml generation
I looked into this for a bit, and I figured out the needed script (example running it on my copy of the repo): 

```
curl $(gh api \
  -H "Accept: application/vnd.github+json" \
  /repos/curino/substrait/collaborators --jq ".[].url") --silent | jq -r '"- Name: \(.name)\n  Association: \(.company)"'                         
- Name: Jesús Camacho Rodríguez
  Association: Microsoft
- Name: Carlo Curino
  Association: Microsoft - CISL
```

I don't know Github Actions well at all, but after a bunch of fussing I can't get it to run (either fails or gets skipped with no explanation). I "think" it has issues with using secrets in the repo, I tried with a couple of options based on [this](https://github.blog/2021-03-11-scripting-with-github-cli/#:~:text=Access%20anything%20with%20gh%20api%20For%20operations%20that,authentication%2C%20parameter%20serialization%2C%20and%20decoding%20JSON%20for%20you.) to no eval.  I suggest we either put the above in a script and run it once in a while by hand to generate `collaborators.yaml` or someone with stronger Github Actions jujitsu skills pick this up.

### PR DESCRIPTION
**Pull request by mbrobbel**
Closes #402.

Fetches collaborators for the governance doc from this repository. I changed
"Affiliation" to "Company" to better match what we fetch from GitHub.

I've also added a weekly trigger for this job to make sure it keeps up to date
even if there are no new changes on the main branch.

This assumes that SMC members have admin permission on this repository.

### CODE REVIEW

**Comment by westonpace** [comment]
Looks like a promising approach. It seems that not everyone has their Github profiles up to date however :)

* @westonpace incorrectly reports Ursa computing (I can fix that).
* @gforsyth lists Capital One (is this correct?)
* @emkornfield 's profile has neither name nor organization.
* It looks like @vibhatha doesn't actually have write permission (I can fix that).
* Organization is missing from several:
  * @richtia 
  * @jinfengni 
  * @JamesRTaylor 
  * @thisisnic

**Comment by gforsyth** [comment]
> @gforsyth lists Capital One (is this correct?)

Used to be :grin: -- fixed now.  Turns out I don't look at my profile very often.

**Comment by westonpace** [comment]
@richtia @jinfengni @JamesRTaylor @thisisnic I would like to move forward with this approach.  If we do then your entry in committers.yaml will be missing organization information (unless you've updated this since Dec. 15th).  I am going to do another review later this week and merge if there are no issues or concerns.

**Comment by richtia** [comment]
> @richtia @jinfengni @JamesRTaylor @thisisnic I would like to move forward with this approach. If we do then your entry in committers.yaml will be missing organization information (unless you've updated this since Dec. 15th). I am going to do another review later this week and merge if there are no issues or concerns.

Mine should be updated.

File: site/data/committers.yaml [hunk 1]
```diff
@@ -1,24 +1,2 @@
...
-- Name: Rich Tia
-  Association: Voltron Data
-- Name: Vibhatha Abeykoon
-  Association: Voltron Data
-- Name: Nic Crane
-  Association: Voltron Data
-- Name: Gil Forsyth
-  Association: Voltron Data
+- Name:
+  Company:
```
Comment (westonpace) [inline site/data/committers.yaml:2]:
```suggestion
# This file is automatically generated in .github/workflows/site.yaml
```

File: site/data/smc.yaml [hunk 2]
```diff
@@ -1,6 +1,2 @@
-- Name: Phillip Cloud
-  Association: Voltron Data
-- Name: Weston Pace
-  Association: Voltron Data
-- Name: Jacques Nadeau
-  Association: Sundeck
+- Name:
+  Company:
```
Comment (westonpace) [inline site/data/smc.yaml:2]:
```suggestion
# This file is automatically generated in .github/workflows/site.yaml
```

**Review by westonpace — COMMENTED** [review]
One small suggestion

**Comment by westonpace** [comment]
I tried running this locally and got the following error:

```
jq: error: syntax error, unexpected INVALID_CHARACTER, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
```

Full failed run: https://github.com/westonpace/substrait/actions/runs/4416495998/jobs/7740865640

**Comment by mbrobbel** [comment]
> I tried running this locally and got the following error:
> 
> ```
> jq: error: syntax error, unexpected INVALID_CHARACTER, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
> ```
> 
> Full failed run: https://github.com/westonpace/substrait/actions/runs/4416495998/jobs/7740865640

Thanks for checking! I pushed a fix and tested in on my fork https://github.com/mbrobbel/substrait/actions/runs/4419172159/jobs/7747258832 ✅

File: .github/workflows/site.yml [hunk 3]
```diff
@@ -34,6 +36,49 @@ jobs:
...
+              company: .[2]
+            }
+          ]' > collaborators.json
+      - name: smc.yaml
+        run: |
+          jq 'map(select(.permission == "ADMIN"))
+            | map({
+                "Name":.name,
+                "Company":.company
+              })' collaborators.json | yq -P > site/data/smc.yaml
```
Comment (gforsyth) [inline .github/workflows/site.yml:74]: I know github actions come with `jq` pre-installed -- is the same true for `yq`?  Or do we need to use https://github.com/frenck/action-setup-yq?

Comment (gforsyth) [inline .github/workflows/site.yml:74]:
```suggestion
              })' collaborators.json | yq -y > site/data/smc.yaml
```

File: .github/workflows/site.yml [hunk 4]
```diff
@@ -34,6 +36,49 @@ jobs:
...
+      - name: committers.yaml
+        run: |
+          jq 'map(select(.permission == "WRITE"))
+            | map({
+                "Name":.name,
+                "Company":.company
+              })' collaborators.json | yq -P > site/data/committers.yaml
```
Comment (gforsyth) [inline .github/workflows/site.yml:81]:
```suggestion
              })' collaborators.json | yq -y > site/data/committers.yaml
```

File: .github/workflows/site.yml (hunk 3 above)
Comment (mbrobbel) [reply .github/workflows/site.yml:74]: It's pre-installed: https://github.com/actions/runner-images/blob/e0371b1f6ff0803f8f938779af7fe6e705e2b4eb/images/linux/Ubuntu2204-Readme.md?plain=1#L108

**Review by gforsyth — COMMENTED** [review]
Tried to run this locally and it works, but with `yq` installed via both `conda-forge` and `nix`, there is no `-P` flag.  I have a `-y` flag to convert json to YAML.

I guess we need to figure out which flag is available on the version we install onto the actions runners

**Comment by gforsyth** [comment]
> Tried to run this locally and it works, but with yq installed via both conda-forge and nix, there is no -P flag. I have a -y flag to convert json to YAML.

I guess this is an artifact of the 3.x to 4.x release and `-P` is definitely available in 4.x

Sorry for the noise!

**Comment by westonpace** [comment]
Hmm, locally it works for me as well.  However, it appears this did not work in production.  Perhaps the token doesn't have sufficient permissions?

**Comment by westonpace** [comment]
Yes, it seems it is only fetching members who have their organization settings for Substrait set to public.

**Comment by gforsyth** [comment]
Ok, well, we were asking committers to PR themselves on to the list.  Now we'll ask them to mark their membership as public instead.

**Pull request merged**